hack/util

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+: ${PREFER_DOCKER=}
+: ${PREFER_LEGACY=}
+
+buildmode="legacy"
+if [ "$PREFER_DOCKER" != "1" ] && buildctl debug workers 2>/dev/null >/dev/null; then 
+	buildmode="buildkit";
+else
+	serverVersion=$(docker info --format '{{.ServerVersion}}')
+	experimental=$(docker info --format '{{.ExperimentalBuild}}')
+	if [ "$PREFER_LEGACY" != "1" ] && ([ "18.09" = $(echo -e "18.09\n$serverVersion" | sort -V | head -n 1) ] || \
+	([ "18.06" = $(echo -e "18.06\n$serverVersion" | sort -V | head -n 1) ] && [ "true" = "$experimental" ] ) || \
```

Comment:
declaring something like this (untested) might help readability:
```
newerThan() { # $1=actual-version $2=minimum wanted version
    [ "$1" = $(echo -e "$1\n$2" | sort -V | head -n 1)]
}
```

Then having `&& newerThan($serverVersion, "18.09") || (newerThan($serverVersion, "18.06") && ["true" = "$exprimental"])` here.

(I hadn't seen `sort -V` before -- very useful!)